From 7b10f20ab69c8cc3f00c56f64ed50e56b7705bbe Mon Sep 17 00:00:00 2001
From: Dmitrii Golovanov <dmitrii.golovanov@intel.com>
Date: Fri, 7 Mar 2025 16:53:47 +0100
Subject: [PATCH] Fix cmake error on in-source build (#1670)

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cdb89fd22b..64a718b80b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -252,7 +252,11 @@ endif()
 if (TBB_FILE_TRIM)
     file(RELATIVE_PATH TBB_RELATIVE_BIN_PATH ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR})
     file(TO_NATIVE_PATH ${CMAKE_SOURCE_DIR} NATIVE_TBB_PROJECT_ROOT_DIR)
-    file(TO_NATIVE_PATH ${TBB_RELATIVE_BIN_PATH} NATIVE_TBB_RELATIVE_BIN_PATH)
+    if (TBB_RELATIVE_BIN_PATH)
+        file(TO_NATIVE_PATH ${TBB_RELATIVE_BIN_PATH} NATIVE_TBB_RELATIVE_BIN_PATH)
+    else()
+        file(TO_NATIVE_PATH ${CMAKE_CURRENT_BINARY_DIR} NATIVE_TBB_RELATIVE_BIN_PATH)
+    endif()
 endif ()
 
 if (TBB_CONTROL_FLOW_GUARD)
