diff --git a/os_dep/os_intfs.c b/os_dep/os_intfs.c
index 597976c..95736b5 100644
--- a/os_dep/os_intfs.c
+++ b/os_dep/os_intfs.c
@@ -22,6 +22,10 @@
 #include <drv_types.h>
 #include <hal_data.h>
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 8, 0))
+#define strlcpy strscpy
+#endif
+
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek Wireless Lan Driver");
 MODULE_AUTHOR("Realtek Semiconductor Corp.");
diff --git a/os_dep/usb_intf.c b/os_dep/usb_intf.c
index 20b827c..db6cec2 100644
--- a/os_dep/usb_intf.c
+++ b/os_dep/usb_intf.c
@@ -172,7 +172,7 @@ static struct rtw_usb_drv usb_drv = {
 	.usbdrv.supports_autosuspend = 1,
 	#endif
 
-	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19))
+	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19)) && (LINUX_VERSION_CODE < KERNEL_VERSION(6, 8, 0))
 	.usbdrv.drvwrap.driver.shutdown = rtw_dev_shutdown,
 	#else
 	.usbdrv.driver.shutdown = rtw_dev_shutdown,
