diff -Naur sysvinit-2.99.orig/src/sulogin.c sysvinit-2.99/src/sulogin.c
--- sysvinit-2.99.orig/src/sulogin.c	2021-02-21 18:02:25.000000000 +0000
+++ sysvinit-2.99/src/sulogin.c	2022-08-05 21:41:42.942846229 +0100
@@ -50,7 +50,6 @@
 #include <sys/ioctl.h>
 #ifdef __linux__
 #  include <sys/statfs.h>
-#  include <sys/mount.h>
 #  include <linux/fs.h>
 #  include <linux/magic.h>
 #  include <linux/major.h>
