diff --git a/meson_options.txt b/meson_options.txt
index 384f984ad57..a2c981a2cc0 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -94,6 +94,14 @@ option(
   description : 'enable gallium vdpau frontend.',
 )
 
+option(
+  'gallium-vc4-neon',
+  type : 'combo',
+  value : 'auto',
+  choices : ['auto', 'disabled'],
+  description : 'enable gallium vc4 optional neon support.',
+)
+
 option(
   'vdpau-libs-path',
   type : 'string',
diff --git a/src/gallium/drivers/vc4/meson.build b/src/gallium/drivers/vc4/meson.build
index 84da9511cdf..76bc90288d1 100644
--- a/src/gallium/drivers/vc4/meson.build
+++ b/src/gallium/drivers/vc4/meson.build
@@ -84,7 +84,7 @@ files_libvc4 = files(
 vc4_c_args = []
 
 libvc4_neon = []
-if host_machine.cpu_family() == 'arm'
+if host_machine.cpu_family() == 'arm' and get_option('gallium-vc4-neon') != 'disabled'
   libvc4_neon = static_library(
     'vc4_neon',
     'vc4_tiling_lt_neon.c',
@@ -93,7 +93,7 @@ if host_machine.cpu_family() == 'arm'
     ],
     c_args : '-mfpu=neon',
   )
-  vc4_c_args += '-DUSE_ARM_ASM'
+  vc4_c_args += '-DVC4_TILING_LT_NEON'
 endif
 
 if dep_simpenrose.found()
diff --git a/src/gallium/drivers/vc4/vc4_tiling.h b/src/gallium/drivers/vc4/vc4_tiling.h
index 5afe32939a8..919b2990ab1 100644
--- a/src/gallium/drivers/vc4/vc4_tiling.h
+++ b/src/gallium/drivers/vc4/vc4_tiling.h
@@ -89,7 +89,7 @@ vc4_load_lt_image(void *dst, uint32_t dst_stride,
                   void *src, uint32_t src_stride,
                   int cpp, const struct pipe_box *box)
 {
-#ifdef USE_ARM_ASM
+#ifdef VC4_TILING_LT_NEON
         if (util_get_cpu_caps()->has_neon) {
                 vc4_load_lt_image_neon(dst, dst_stride, src, src_stride,
                                        cpp, box);
@@ -105,7 +105,7 @@ vc4_store_lt_image(void *dst, uint32_t dst_stride,
                    void *src, uint32_t src_stride,
                    int cpp, const struct pipe_box *box)
 {
-#ifdef USE_ARM_ASM
+#ifdef VC4_TILING_LT_NEON
         if (util_get_cpu_caps()->has_neon) {
                 vc4_store_lt_image_neon(dst, dst_stride, src, src_stride,
                                         cpp, box);
