diff --git a/src/util/compiler.h b/src/util/compiler.h
index d184ad455af..b5c56807acc 100644
--- a/src/util/compiler.h
+++ b/src/util/compiler.h
@@ -36,6 +36,16 @@
 
 #include <assert.h>
 
+/*
+ * C11 static_assert() macro
+ * assert.h only defines that name for C11 and above
+ */
+#if !defined(__cplusplus)
+#ifndef static_assert
+#define static_assert _Static_assert
+#endif
+#endif
+
 #include "util/macros.h"
 
 
diff --git a/src/util/macros.h b/src/util/macros.h
index 4175a795a8f..f57cc065b61 100644
--- a/src/util/macros.h
+++ b/src/util/macros.h
@@ -28,6 +28,7 @@
 #include <stddef.h>
 #include <stdint.h>
 #include <stdio.h>
+#include "util/compiler.h"
 
 /* Compute the size of an array */
 #ifndef ARRAY_SIZE