diff --git a/gif.c b/gif.c
index e13b060..9868df5 100644
--- a/gif.c
+++ b/gif.c
@@ -48,7 +48,7 @@ int fh_gif_id(char *name)
     return(0);
 }
 
-inline void m_rend_gif_decodecolormap(unsigned char *cmb,unsigned char *rgbb,ColorMapObject *cm,int s,int l, int transparency)
+static inline void m_rend_gif_decodecolormap(unsigned char *cmb,unsigned char *rgbb,ColorMapObject *cm,int s,int l, int transparency)
 {
     GifColorType *cmentry;
     int i;
