diff -Naur lua-5.5.1.oirg/src/lua.c lua-5.5.1/src/lua.c
--- lua-5.5.1.oirg/src/lua.c	2026-07-24 11:23:07.000000000 +0100
+++ lua-5.5.1/src/lua.c	2026-08-08 18:52:11.468210116 +0100
@@ -438,6 +438,14 @@
 
 #define lua_stdin_is_tty()	_isatty(_fileno(stdin))
 
+#elif defined(LUA_USE_LINENOISE)
+
+#include <linenoise.h>
+#define lua_initreadline(L)    ((void)L)
+#define lua_readline(L,b,p)    ((void)L, ((b)=linenoise(p)) != NULL)
+#define lua_saveline(L,line)   ((void)L, linenoiseHistoryAdd(line))
+#define lua_freeline(L,b)      ((void)L, free(b))
+
 #else				/* }{ */
 
 /* ISO C definition */
