From f8468c316fbe2c559a346de212cb7ad6ee1507a8 Mon Sep 17 00:00:00 2001
From: Tobias Hieta <tobias@hieta.se>
Date: Tue, 8 Aug 2023 13:16:44 +0200
Subject: [PATCH] Bump version to 17.0.0-rc2

---
 llvm/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index a9b85b433c75ae..2eb188d26ae82a 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -18,7 +18,7 @@ if(NOT DEFINED LLVM_VERSION_PATCH)
   set(LLVM_VERSION_PATCH 0)
 endif()
 if(NOT DEFINED LLVM_VERSION_SUFFIX)
-  set(LLVM_VERSION_SUFFIX -rc1)
+  set(LLVM_VERSION_SUFFIX -rc2)
 endif()
 
 if (NOT PACKAGE_VERSION)
