diff --git a/common/compat.h b/common/compat.h
index 20b050a..782307b 100644
--- a/common/compat.h
+++ b/common/compat.h
@@ -308,7 +308,7 @@ void *     reallocarray     (void *ptr,
 #ifdef HAVE_STDBOOL_H
 #include <stdbool.h>
 #else
-typedef enum { false, true } bool;
+#include <stdbool.h>
 #endif
 
 #ifndef HAVE_STRCONCAT
diff --git a/p11-kit/export-object.c b/p11-kit/export-object.c
index ca61b7a..f624155 100644
--- a/p11-kit/export-object.c
+++ b/p11-kit/export-object.c
@@ -234,7 +234,7 @@ export_pubkey_ec (const ExportData *data,
 	unsigned char *der = NULL;
 
 	attrs = p11_attrs_buildn (NULL, template, p11_attrs_count (template));
-	return_val_if_fail (attrs, false);
+	return_val_if_fail (attrs, NULL);
 
 	if (p11_kit_iter_load_attributes (data->iter,
 					  attrs,
diff --git a/trust/x509.c b/trust/x509.c
index b3d4296..464b996 100644
--- a/trust/x509.c
+++ b/trust/x509.c
@@ -122,7 +122,7 @@ p11_x509_parse_subject_key_identifier  (p11_dict *asn1_defs,
 	unsigned char *keyid;
 	asn1_node ext;
 
-	return_val_if_fail (keyid_len != NULL, false);
+	return_val_if_fail (keyid_len != NULL, NULL);
 
 	ext = p11_asn1_decode (asn1_defs, "PKIX1.SubjectKeyIdentifier", ext_der, ext_len, NULL);
 	if (ext == NULL)
@@ -259,7 +259,7 @@ p11_x509_parse_directory_string (const unsigned char *input,
 	return_val_if_fail (ret == ASN1_SUCCESS, NULL);
 
 	octet_len = asn1_get_length_der (input + tag_len, input_len - tag_len, &len_len);
-	return_val_if_fail (octet_len >= 0, false);
+	return_val_if_fail (octet_len >= 0, NULL);
 	return_val_if_fail (tag_len + len_len + octet_len == input_len, NULL);
 
 	octets = input + tag_len + len_len;
