diff -Naur ifupdown.orig/execute.c ifupdown/execute.c
--- ifupdown.orig/execute.c	2024-08-20 03:02:23.000000000 +0100
+++ ifupdown/execute.c	2024-12-21 21:16:32.830761873 +0000
@@ -104,7 +104,7 @@
 	*ppch++ = setlocalenv("%s=%s", "MODE", mode);
 	*ppch++ = setlocalenv("%s=%s", "PHASE", phase);
 	*ppch++ = setlocalenv("%s=%s", "VERBOSITY", verbose ? "1" : "0");
-	*ppch++ = setlocalenv("%s=%s", "PATH", PKGLIBDIR ":/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin");
+	*ppch++ = setlocalenv("%s=%s", "PATH", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin");
 	if (allow_class || do_all)
 		*ppch++ = setlocalenv("%s=%s", "CLASS", allow_class ? allow_class : "auto");
 	*ppch = NULL;
