diff -Naur ifupdown-0.8.45.orig/header.h ifupdown-0.8.45/header.h
--- ifupdown-0.8.45.orig/header.h	2026-01-07 11:05:44.000000000 +0000
+++ ifupdown-0.8.45/header.h	2026-01-08 20:42:28.334501180 +0000
@@ -100,7 +100,7 @@
 #define RUN_DIR "/run/network/"
 #endif
 
-#define EXECUTE_LOCAL_PATHLIST (PKGLIBDIR ":/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin")
+#define EXECUTE_LOCAL_PATHLIST ("/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin")
 
 #ifndef LO_IFACE
 #define LO_IFACE "lo"
