diff -Naur ifupdown.orig/Makefile ifupdown/Makefile
--- ifupdown.orig/Makefile	2024-08-20 03:02:23.000000000 +0100
+++ ifupdown/Makefile	2024-12-21 21:24:05.364201686 +0000
@@ -26,8 +26,8 @@
 install :
 	install -m 0755 -d     ${BASEDIR}/usr/sbin
 	install -m 0755 ifup   ${BASEDIR}/usr/sbin
-	ln -s /usr/sbin/ifup ${BASEDIR}/usr/sbin/ifdown
-	ln -s /usr/sbin/ifup ${BASEDIR}/usr/sbin/ifquery
+	ln -sf /usr/sbin/ifup ${BASEDIR}/usr/sbin/ifdown
+	ln -sf /usr/sbin/ifup ${BASEDIR}/usr/sbin/ifquery
 	install -D -m 0755 settle-dad.sh $(BASEDIR)$(PKGLIBDIR)/settle-dad.sh
 	install -D -m 0755 wait-for-ll6.sh $(BASEDIR)$(PKGLIBDIR)/wait-for-ll6.sh
 	install -D -m 0755 wait-online.sh $(BASEDIR)$(PKGLIBDIR)/wait-online.sh
