diff -Naur wayland-1.19.0.orig/configure.ac wayland-1.19.0/configure.ac
--- wayland-1.19.0.orig/configure.ac	2021-01-27 16:49:04.000000000 +0000
+++ wayland-1.19.0/configure.ac	2021-07-17 12:54:48.439125185 +0100
@@ -77,6 +77,13 @@
 	      [],
 	      [enable_libraries=yes])
 
+AC_ARG_ENABLE([tests],
+	      [AC_HELP_STRING([--disable-tests],
+			      [Disable compilation of test programs])],
+	      [],
+	      [enable_tests=yes])
+
+
 AC_ARG_WITH([host-scanner],
               [AC_HELP_STRING([--with-host-scanner],
                               [Use installed wayland-scanner from host PATH during build])],
@@ -99,6 +106,8 @@
 
 AM_CONDITIONAL(ENABLE_LIBRARIES, test "x$enable_libraries" = xyes)
 
+AM_CONDITIONAL(ENABLE_TESTS, test "x$enable_tests" = xyes)
+
 AC_ARG_WITH(icondir, [  --with-icondir=<dir>    Look for cursor icons here],
 		     [  ICONDIR=$withval],
 		     [  ICONDIR=${datadir}/icons])
diff -Naur wayland-1.19.0.orig/Makefile.am wayland-1.19.0/Makefile.am
--- wayland-1.19.0.orig/Makefile.am	2021-01-27 16:49:03.000000000 +0000
+++ wayland-1.19.0/Makefile.am	2021-07-17 12:54:48.439125185 +0100
@@ -162,6 +162,7 @@
 include_HEADERS += egl/wayland-egl-backend.h
 pkgconfig_DATA += egl/wayland-egl-backend.pc
 
+if ENABLE_TESTS
 built_test_programs =				\
 	array-test				\
 	client-test				\
@@ -295,6 +296,7 @@
 
 exec_fd_leak_checker_SOURCES = tests/exec-fd-leak-checker.c
 exec_fd_leak_checker_LDADD = libtest-helpers.la
+endif
 
 EXTRA_DIST += tests/scanner-test.sh			\
 	protocol/tests.xml				\
diff -Naur wayland-1.19.0.orig/meson.build wayland-1.19.0/meson.build
--- wayland-1.19.0.orig/meson.build	2021-01-27 16:49:04.000000000 +0000
+++ wayland-1.19.0/meson.build	2021-07-17 12:56:28.347994376 +0100
@@ -91,7 +91,9 @@
 if get_option('libraries')
 	subdir('cursor')
 	subdir('egl')
-	subdir('tests')
+	if get_option('tests')
+		subdir('tests')
+	endif
 	if get_option('documentation')
 		subdir('doc')
 	endif
diff -Naur wayland-1.19.0.orig/meson_options.txt wayland-1.19.0/meson_options.txt
--- wayland-1.19.0.orig/meson_options.txt	2021-01-27 16:49:04.000000000 +0000
+++ wayland-1.19.0/meson_options.txt	2021-07-17 12:54:48.439125185 +0100
@@ -6,6 +6,10 @@
   description: 'Compile wayland-scanner binary',
   type: 'boolean',
   value: 'true')
+option('tests',
+  description: 'Compile Wayland tests',
+  type: 'boolean',
+  value: 'true')
 option('documentation',
   description: 'Build the documentation (requires Doxygen, dot, xmlto, xsltproc)',
   type: 'boolean',
