Fix python detection

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr xbmc-20.0rc1-Nexus.orig/cmake/modules/FindPython.cmake xbmc-20.0rc1-Nexus/cmake/modules/FindPython.cmake
--- xbmc-20.0rc1-Nexus.orig/cmake/modules/FindPython.cmake	2022-12-10 05:39:56.000000000 +0100
+++ xbmc-20.0rc1-Nexus/cmake/modules/FindPython.cmake	2022-12-10 10:43:25.796686397 +0100
@@ -60,7 +60,7 @@
 
 find_package(Python3 ${VERSION} ${EXACT_VER} COMPONENTS Development)
 if(CORE_SYSTEM_NAME STREQUAL linux)
-  if(HOST_CAN_EXECUTE_TARGET)
+  if(HOST_CAN_EXECUTE_TARGET OR DEFINED PYTHON_EXECUTABLE)
     find_package(Python3 ${VERSION} ${EXACT_VER} COMPONENTS Interpreter)
   else()
     find_package(Python3 COMPONENTS Interpreter)
