diff --git a/src/trans.c b/src/trans.c
index 3603436..cb2b070 100644
--- a/src/trans.c
+++ b/src/trans.c
@@ -1,6 +1,9 @@
 /*  $Id: trans.c,v 1.19 2004/11/22 02:28:21 mgrouch Exp $  */
 
 #include <config.h>
+
+#include <libxml/xmlsave.h>
+
 #include "trans.h"
 #include "xmlstar.h"
 
diff --git a/src/xml_edit.c b/src/xml_edit.c
index a588789..1334b9d 100644
--- a/src/xml_edit.c
+++ b/src/xml_edit.c
@@ -466,7 +466,6 @@ edOutput(const char* filename, edOptions g_ops)
     {
         cleanupNSArr(ns_arr);
         xmlCleanupParser();
-        xmlCleanupGlobals();
         exit(EXIT_BAD_FILE);
     }
 
@@ -721,6 +720,5 @@ edMain(int argc, char **argv)
 
     cleanupNSArr(ns_arr);
     xmlCleanupParser();
-    xmlCleanupGlobals();
     return 0;
 }
diff --git a/src/xml_format.c b/src/xml_format.c
index 77c0cdf..958fc0c 100644
--- a/src/xml_format.c
+++ b/src/xml_format.c
@@ -35,6 +35,7 @@ THE SOFTWARE.
 #include <libxml/xmlmemory.h>
 #include <libxml/debugXML.h>
 #include <libxml/xmlIO.h>
+#include <libxml/xmlsave.h>
 #include <libxml/HTMLtree.h>
 #include <libxml/xinclude.h>
 #include <libxml/xpath.h>
