diff --git a/configure.ac b/configure.ac
index 879e6a2d432..7bc049863c3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5433,7 +5433,7 @@ fi]))
 dnl if ac_cv_func_getaddrinfo
 ])
 
-if test "$ac_cv_func_getaddrinfo" = no -o "$ac_cv_buggy_getaddrinfo" = yes
+if test "$ac_cv_func_getaddrinfo" = no || test "$cross_compiling" != "yes" -a "$ac_cv_buggy_getaddrinfo" = yes
 then
   AS_VAR_IF([ipv6], [yes], [
     AC_MSG_ERROR([m4_normalize([
