diff -Naur libedit-20251016-3.1.orig/configure.ac libedit-20251016-3.1/configure.ac
--- libedit-20251016-3.1.orig/configure.ac	2025-10-16 18:40:41.000000000 +0100
+++ libedit-20251016-3.1/configure.ac	2025-10-18 09:28:06.486007645 +0100
@@ -49,7 +49,9 @@
   [AC_CHECK_LIB(curses, tgetent,,
     [AC_CHECK_LIB(termcap, tgetent,,
       [AC_CHECK_LIB(tinfo, tgetent,,
-        [AC_MSG_ERROR([libncurses, libcurses, libtermcap or libtinfo is required!])]
+        [AC_CHECK_LIB(tinfow, tgetent,,
+          [AC_MSG_ERROR([libncurses, libcurses, libtermcap or libtinfo is required!])]
+        )]
       )]
     )]
   )]
