diff --git a/configure.ac b/configure.ac
index 420421d8aa..efad51ad31 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3308,6 +3308,7 @@ AS_IF([test "${enable_wayland}" = "yes"], [
     ])
 
     AC_MSG_CHECKING([for the Wayland scanner])
+    AS_VAR_SET_IF(WAYLAND_SCANNER, [],[
     PKG_CHECK_EXISTS([wayland-scanner], [
       WAYLAND_SCANNER="$(${PKG_CONFIG} wayland-scanner --variable wayland_scanner)"
       AC_MSG_RESULT([${WAYLAND_SCANNER}])
@@ -3315,6 +3316,7 @@ AS_IF([test "${enable_wayland}" = "yes"], [
       AC_MSG_RESULT([not found])
       AC_MSG_ERROR([$(${PKG_CONFIG} --print-errors wayland-scanner)])
     ])
+    ])
 
     have_wayland="yes"
 
