diff --git a/modules/video_filter/opencv_example.cpp b/modules/video_filter/opencv_example.cpp
index a7a0bd7821..23483f0974 100644
--- a/modules/video_filter/opencv_example.cpp
+++ b/modules/video_filter/opencv_example.cpp
@@ -197,7 +197,7 @@ static picture_t *Filter( filter_t *p_filter, picture_t *p_pic )
             pt2.x = (r->x+r->width)*scale;
             pt1.y = r->y*scale;
             pt2.y = (r->y+r->height)*scale;
-            cvRectangle( p_img[0], pt1, pt2, CV_RGB(0,0,0), 3, 8, 0 );
+            cvRectangle( p_img[0], pt1, pt2, cvScalar(0,0,0,0), 3, 8, 0 );
 
             *(CvRect*)(&(p_sys->event_info.p_region[i])) = *r;
             p_sys->event_info.p_region[i].i_id = p_sys->i_id++;
