diff --git a/CMake/catch2-with-main.pc.in b/CMake/catch2-with-main.pc.in
index eeac3e35..69a790bb 100644
--- a/CMake/catch2-with-main.pc.in
+++ b/CMake/catch2-with-main.pc.in
@@ -1,10 +1,10 @@
-prefix=@CMAKE_INSTALL_PREFIX@
-libdir=${prefix}/@lib_dir@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 pkg_version=@Catch2_VERSION@
 
-Name: Catch2 with main function
+Name: Catch2-With-Main
 Description: A modern, C++-native test framework for C++14 and above (links in default main)
-URL: https://github.com/catchorg/Catch2
 Version: ${pkg_version}
 Requires: catch2 = ${pkg_version}
-Libs: -L${libdir} -l@lib_name@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lCatch2Main
diff --git a/CMake/catch2.pc.in b/CMake/catch2.pc.in
index 74e22c15..bd1c95a1 100644
--- a/CMake/catch2.pc.in
+++ b/CMake/catch2.pc.in
@@ -1,11 +1,11 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-includedir=${prefix}/@include_dir@
-libdir=${prefix}/@lib_dir@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 
 Name: Catch2
 Description: A modern, C++-native, test framework for C++14 and above
 URL: https://github.com/catchorg/Catch2
 Version: @Catch2_VERSION@
 Cflags: -I${includedir}
-Libs: -L${libdir} -l@lib_name@
+Libs: -L${libdir} -lCatch2
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1676ee79..90c7b88c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,41 +176,23 @@ if(NOT_SUBPROJECT)
     "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig"
     CACHE PATH "Path where catch2.pc is installed"
   )
-
-  # Generate the pkg-config files
-  # To understand the script below, you have to understand that it works in two steps.
-  # 1) A CMake script is generated at configuration time
-  # 2) It is executed at install time.
-  # And both of these have access to different parts of the information we need.
-  #
-  # Further, the variables before "[[" are expanded at configuration time,
-  # while the ones inside the [[]] block are expanded at script execution (install) time.
-  string(
-    JOIN "\n"
-      install_script
-       "set(install_pkgconfdir \"${PKGCONFIG_INSTALL_DIR}\")"
-       "set(impl_pc_file \"${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2.pc.in\")"
-       "set(main_pc_file \"${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2-with-main.pc.in\")"
-       "set(Catch2_VERSION ${Catch2_VERSION})"
-       "set(include_dir \"${CMAKE_INSTALL_INCLUDEDIR}\")"
-       "set(lib_dir \"${CMAKE_INSTALL_LIBDIR}\")"
-       [[
-         set(lib_name "$<TARGET_FILE_BASE_NAME:Catch2>")
-         configure_file(
-           "${impl_pc_file}"
-           "${CMAKE_INSTALL_PREFIX}/${install_pkgconfdir}/catch2.pc"
-           @ONLY
-         )
-
-         set(lib_name "$<TARGET_FILE_BASE_NAME:Catch2WithMain>")
-         configure_file(
-           "${main_pc_file}"
-           "${CMAKE_INSTALL_PREFIX}/${install_pkgconfdir}/catch2-with-main.pc"
-           @ONLY
-         )
-       ]]
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2.pc.in
+    ${CMAKE_CURRENT_BINARY_DIR}/catch2.pc
+    @ONLY
+  )
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2-with-main.pc.in
+    ${CMAKE_CURRENT_BINARY_DIR}/catch2-with-main.pc
+    @ONLY
+  )
+  install(
+    FILES
+      "${CMAKE_CURRENT_BINARY_DIR}/catch2.pc"
+      "${CMAKE_CURRENT_BINARY_DIR}/catch2-with-main.pc"
+    DESTINATION
+      ${PKGCONFIG_INSTALL_DIR}
   )
-  install(CODE "${install_script}")
 
   set(CPACK_PACKAGE_CONTACT "https://github.com/catchorg/Catch2/")
 
