diff --git a/configure.ac b/configure.ac
index 5d88da05..c68169f4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -84,7 +84,7 @@ fi
 
 
 # intltool
-IT_PROG_INTLTOOL([0.40.0])
+#IT_PROG_INTLTOOL([0.40.0])
 
 dnl libgtk version
 AC_ARG_WITH(gtk,
@@ -322,14 +322,7 @@ AC_CONFIG_FILES([
     src/modules/Makefile
     src/fm-version.h
     data/Makefile
-    data/libfm-pref-apps.1
-    data/lxshortcut.1
     data/ui/Makefile
-    po/Makefile.in
-    docs/Makefile
-    docs/reference/Makefile
-    docs/reference/libfm/Makefile
-    docs/reference/libfm/version.xml
     libfm.pc
     libfm-gtk.pc
     libfm-gtk3.pc
